lib/appium.js

```diff
@@ -321,10 +321,30 @@ class AppiumDriver extends BaseDriver {
 
       let runningDriversData, otherPendingDriversData;
       const d = new InnerDriver(this.args);
+
+      // We want to assign security values directly on the driver. The driver
+      // should not read security values from `this.opts` because those values
+      // could have been set by a malicious user via capabilities, whereas we
+      // want a guarantee the values were set by the appium server admin
       if (this.args.relaxedSecurityEnabled) {
-        log.info(`Applying relaxed security to '${InnerDriver.name}' as per server command line argument`);
+        log.info(`Applying relaxed security to '${InnerDriver.name}' as per ` +
+                 `server command line argument. All insecure features will be ` +
+                 `enabled unless explicitly disabled by --deny-insecure`);
         d.relaxedSecurityEnabled = true;
       }
+
+      if (!_.isEmpty(this.args.denyInsecure)) {
+        log.info(`Explicitly preventing use of insecure features ` +
+                 `${JSON.stringify(this.args.allowInsecure)} for this server`);
```

Comment:
This log line will be hard to read for anything but the simplest case. It would make more sense, I think, to have them listed on their own line, so it is abundantly clear what is and is not available.